src/nupic/algorithms/ClassifierResult.cpp

```diff
@@ -48,15 +49,11 @@ namespace nupic
       vector<Real64>* ClassifierResult::createVector(Int step, UInt size,
                                                 Real64 value)
       {
-        vector<Real64>* v;
         map<Int, vector<Real64>*>::const_iterator it = result_.find(step);
-        if (it != result_.end())
-        {
-          v = it->second;
-        } else {
-          v = new vector<Real64>(size, value);
-          result_.insert(pair<Int, vector<Real64>*>(step, v));
-        }
+        NTA_CHECK(it == result_.end())
+          << "The ClassifierResult cannot be reused!";
```

Comment:
Maybe just use

~~~c++
NTA_CHECK(result_.count(step) == 0);
~~~

and get rid of the iterator.